File: lib/features/products/domain/product.dart
class Product {
  const Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.category,
    this.imageUrl,
  });

  final String id;
  final String name;
  final String description;
  final double price;
  final String category;
  final String? imageUrl;
}

File: apps/storefront/src/features/catalog/domain/product.ts
export type ProductImage = {
  id: string;
  variantId?: string;
  url: string;
  altText: string;
  isPrimary: boolean;
};

export type ProductVariant = {
  id: string;
  sku: string;
  name: string;
  price: number;
  compareAtPrice?: number;
  currency: "MYR";
  availableStock: number;
  inStock: boolean;
  optionValues: Record<string, string>;
};

export type ProductDetails = {
  material?: string;
  dimensions?: string;
  care?: string;
  highlights: string[];
  specifications: Record<string, string>;
};

export type Product = {
  id: string;
  variantId: string;
  slug: string;
  name: string;
  description: string;
  details: ProductDetails;
  colorSwatches: Record<string, string>;
  price: number;
  currency: "MYR";
  category: string;
  badge?: string;
  inStock: boolean;
  images: ProductImage[];
  variants: ProductVariant[];
};
